personalize.admin.inc

```diff
@@ -85,6 +85,89 @@ function personalize_admin_form_submit($form, &$form_state) {
 }
 
 /**
+ * Form for changing the status of an agent.
+ *
+ * @param $agent_name
+ *   THe name of the agent the form is for.
+ * @param array $agent_status
+ *   The current status of the agent.
+ * @return array
+ *   Array representing the status change form.
+ */
+function personalize_status_change_form($form, &$form_state, $agent_name, $agent_status) {
+  $friendly_statuses = personalize_get_agent_status_map();
+  $form = array();
+  $form['agent_name'] = array(
+    '#type' => 'value',
+    '#value' => $agent_name,
+  );
+  switch($agent_status) {
```

Comment:
Seems like we could use personalize_allowed_status_transitions to dynamically determine this.  Otherwise the transitions are effectively defined in two places - here and the transitions function (and some of the older functions from earlier too).